_functions.R

```diff
@@ -42,15 +45,19 @@ aggregate_benchmark <- function(name, iterations, task) {
     }
   })
 
-  times[[name]] <<- aggregate_time
+  times[[name]] <<- list(time=aggregate_time, parallelism=NA)
 }
 
 benchmark_end <- function() {
-  df <- do.call(rbind, unname(lapply(times, function(proc_time) {
+  df <- do.call(rbind, unname(lapply(times, function(list) {
+    proc_time <- list[[1]]
+    parallelism <- as.integer(list[[2]])
     pt <- summary(proc_time)
-    data.frame(user = pt[[1]], system = pt[[2]], elapsed = pt[[3]])
+    data.frame(user = pt[[1]], system = pt[[2]], elapsed = pt[[3]], parallelism = parallelism)
   })))
-  df <- cbind(data.frame(task = names(times), stringsAsFactors = FALSE), df)
+
+  # strip off the added parallelism in the name when creating the final data frame
+  df <- cbind(data.frame(task = str_replace(names(times), "(.*)\\(([\\d])\\)$", "\\1"), stringsAsFactors = FALSE), df)
```

Comment:
Let's use `sub` or `gsub` here instead of `str_replace` so we don't need a dependency on stringr. stringr depends on stringi, and stringi is quite painful to build from source.

Reply:
Done.